core/src/main/scala/akka/persistence/r2dbc/internal/ContinuousQuery.scala

```diff
@@ -150,8 +153,12 @@ final private[r2dbc] class ContinuousQuery[S, T](
                   next()
                 }
             })
+            val sourceWithHeartbeat = heartbeat(newState) match {
+              case None    => source
+              case Some(h) => Source.single(h).concat(source)
```

Comment:
maybe a little strange to emit it before a real query, but I didn't see an easy way to do it after a query with a fully updated state